Gems/Atom/RHI/Metal/Code/Source/RHI/Scope.cpp

```diff
@@ -93,17 +93,23 @@ namespace AZ
             int colorAttachmentIndex = 0;
             AZStd::unordered_map<RHI::AttachmentId, ResolveAttachmentData> attachmentsIndex;
             
-            for (const RHI::ImageScopeAttachment* scopeAttachment : GetImageAttachments())
+            for (RHI::ImageScopeAttachment* scopeAttachment : GetImageAttachments())
             {
                 m_isWritingToSwapChainScope = scopeAttachment->IsSwapChainAttachment() && scopeAttachment->HasUsage(RHI::ScopeAttachmentUsage::RenderTarget);
                 if(m_isWritingToSwapChainScope)
                 {
-                    //Check if the scope attachment for the next scope is going to capture the frame.
+                    //Check if the swapchain scope attachment will be copied to be read in the upcoming passes
                     //We can use this information to cache the swapchain texture for reading purposes.
-                    const RHI::ScopeAttachment* frameCaptureScopeAttachment = scopeAttachment->GetNext();
-                    if(frameCaptureScopeAttachment)
+                    RHI::ScopeAttachment* frameCaptureScopeAttachment = scopeAttachment;
+                    while(frameCaptureScopeAttachment)
                     {
-                        m_isSwapChainAndFrameCaptureEnabled = frameCaptureScopeAttachment->HasAccessAndUsage(RHI::ScopeAttachmentUsage::Copy, RHI::ScopeAttachmentAccess::Read);
+                        frameCaptureScopeAttachment = frameCaptureScopeAttachment->GetNext();
```

Comment:
This is skipping the first scopeAttachment. Should the advancement be done after checking HasAccessAndUsage?

Reply:
Skipping the first scopeAttachment is intentional. Basically the idea here is that when we are in the CopyToSwapchain scope will ask the driver for swap chain texture but we need to know if this swap chain texture ever be read as it changes the way we request the swap chain texture. So we check all the scopes next in line to see if any of them will be reading the swap chain and if we do we break out after caching the information. I can add a comment here to explain this better